File: server/models/Task.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var taskSchema = new Schema({
    //userID: { type: String, required: true },
    //_id: Schema.Types.ObjectId,
    userId: Schema.Types.ObjectId,
    taskTitle: { type: String },
    taskDescription: { type: String },
    importanceLevel: { type: String },
    deadline: { type: Date },   // format example "2019-11-12T14:39:00.000Z"
    //reminder: { type: Schema.Types.ObjectId, ref: 'Reminder' }
});

module.exports = mongoose.model('Task', taskSchema);

File: Android/app/src/main/java/se/chalmers/cse/dit341/group22/model/Task.java
package se.chalmers.cse.dit341.group22.model;

import java.util.Date;

// My Task Class. Used by GSON to transfer JSON directly to Java Object
public class Task {

    private String _id;
    private String userId;
    private String taskTitle;
    private String taskDescription;
    private String importanceLevel;
    private Date deadline;

    public Task() {
    }

    public String getUserId() {
        return userId;
    }

    public String getImportanceLevel() {
        return importanceLevel;
    }

    public String getId() {
        return _id;
    }

    public Date getDeadline() {
        return deadline;
    }

    public String getTaskDescription() {
        return taskDescription;
    }

    public String getTaskTitle() {
        return taskTitle;
    }

    public void setImportanceLevel(String importanceLevel) {
        this.importanceLevel = importanceLevel;
    }

    public void setDeadline(Date deadline) {
        this.deadline = deadline;
    }

    public void setTaskDescription(String taskDescription) {
        this.taskDescription = taskDescription;
    }

    public void setTaskTitle(String taskTitle) {
        this.taskTitle = taskTitle;
    }
}
